sway-lib-std/src/u128.sw

```diff
@@ -331,3 +331,58 @@ impl Exponentiate for U128 {
         acc
     }
 }
+
+impl Root for U128 {
+    /// Newton's method as in https://en.wikipedia.org/wiki/Integer_square_root#Algorithm_using_Newton's_method
+    fn sqrt(self) -> Self {
+        let zero = ~U128::from(0, 0);
+        let two = ~U128::from(0, 2);
+        let mut x0 = self / two;
+        let mut s = self;
+
+        if x0 != zero {
+            let mut x1 = (x0 + s / x0) / two;
+
+            while x1 < x0 {
+                x0 = x1;
+                x1 = (x0 + self / x0) / two;
+            }
+
+            return x0;
+        } else {
+            return s;
+        }
+    }
+}
+
+impl BinaryLogarithm for U128 {
+    /// As binary form stores a number as sum of powers of 2
+    /// finding logarithm which is the maximum power of 2 
+    /// such that 2 to that power is smeller of equal to the
+    /// original value, and 2 to that power + 1 is already 
+    /// bigger
+    /// In case the value is smaller 2 ^ 64 we could just use
+    /// regular log with the base of 2, based on the opcode
+    /// Otherwise we can find the highest bit by taking
+    /// the regular log of the upper part and add 64
```

Comment:
```suggestion
    /// log2 of `x` is the largest `n` such that `2^n <= x < 2^(n+1)`.
    /// 
    /// * If `x` is smaller than `2^64`, we could just rely on the `log` method by setting
    /// the base to 2.
    /// * Otherwise, we can find the highest non-zero bit by taking the regular log of the upper
    /// part of the `U128`, and then add 64.
```